### PR TITLE
itest+lntest: make sure states are cleaned when tests end

### PR DESCRIPTION
**Pull request by yyforyongyu**
Fixed a case where the cleanup is missed and validated the state when the tests end.

### CODE REVIEW

**Comment by yyforyongyu** [comment]
hmm should this be based on master or `0.16.1`? I guess on master since other branches need this fix?

File: lntest/harness.go [hunk 1]
```diff
@@ -433,8 +430,18 @@ func (h *HarnessTest) cleanupStandbyNode(hn *node.HarnessNode) {
 	// Delete all payments made from this test.
 	hn.RPC.DeleteAllPayments()
 
-	// Finally, check the node is in a clean state for the following tests.
-	h.validateNodeState(hn)
+	// Check the node's current state with timeout.
+	//
+	// NOTE: we need to do this is a `wait` because it takes some time for
```
Comment (ellemouton) [inline lntest/harness.go:435]: s/is/as

File: itest/lnd_htlc_test.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ func testLookupHtlcResolution(ht *lntest.HarnessTest) {
 	cp := ht.OpenChannel(
 		alice, carol, lntest.OpenChannelParams{Amt: chanAmt},
 	)
+	defer ht.CloseChannel(alice, cp)
```
Comment (ellemouton) [inline itest/lnd_htlc_test.go:27]: shweet

File: lntest/harness.go (hunk 1 above)
Comment (ellemouton) [reply lntest/harness.go:435]: or s/is/in

**Review by ellemouton — APPROVED** [review]
Noice 🔥

File: lntest/harness.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/harness.go]: fixed🙈

File: lntest/harness.go [hunk 3]
```diff
@@ -433,8 +430,18 @@ func (h *HarnessTest) cleanupStandbyNode(hn *node.HarnessNode) {
...
 	hn.RPC.DeleteAllPayments()
 
-	// Finally, check the node is in a clean state for the following tests.
-	h.validateNodeState(hn)
+	// Check the node's current state with timeout.
+	//
+	// NOTE: we need to do this in a `wait` because it takes some time for
+	// the node to update its internal state. Once the RPCs are synced we
+	// can then remove this wait.
+	wait.NoError(func() error {
```
Comment (ellemouton) [inline lntest/harness.go:438]: just gotta check the error returned here still

**Review by guggero — APPROVED** [review]
LGTM pending @ellemouton's comments :tada: 

Since we don't know if RC3 will be the final one, should we base this on `0-16-1-staging` instead so we can go ahead and merge it?

**Comment by yyforyongyu** [comment]
Changed the base!

**Pull request merged**